src/client/pages/CountryHome/Repository/Panel/InputFieldFile/InputFieldFile.tsx

```diff
@@ -1,20 +1,30 @@
 import React from 'react'
 import { useTranslation } from 'react-i18next'
 
+import { useRepositoryItem } from 'client/store/ui/repository'
 import FileUpload from 'client/components/FileUpload'
 
+import { useOnRemoveFile } from './hooks/useOnRemoveFile'
+
 const FileInputField: React.FC = () => {
+  const onRemoveFile = useOnRemoveFile()
+  const repositoryItem = useRepositoryItem()
   const { t } = useTranslation()
 
-  const label = `common.file`
   const id = `repository_form-input-file`
+  const label = `common.file`
 
   return (
     <div className="repository-form__input-field">
       <label htmlFor={id} className="repository-form__label">
         {t(label)}
       </label>
       <FileUpload id={id} />
+      {repositoryItem?.fileUuid && (
```

Comment:
I will revisit a bit the fileuploader next PR if this is ok
This might be a bit misleading, but the functionality of the button is not related the UploadedFiles or `public.files` but to `RepositoryItem`

The functionality removes the `repositoryItem.fileUuid`

Case 1:
Delete button is shown:
We open the list of repository items in the platform
We click on pen icon
We click the delete icon
We see an error, either a link or a file is required

Case 2:
We add a new repository item
We upload a file
We see the delete icon
We add a link
We see an error, only one is required file vs link
We click the delete icon


in short: it deletes the fileUuid reference

Reply:
I am thinking of introducing `FileUpload.onDeleteFile`  with `onDeleteFile(uuid: string)` prop but I am not 100% sure on use case yet